spec/workers/purge_worker_spec.rb

```diff
@@ -0,0 +1,92 @@
+require "rails_helper"
+
+RSpec.describe PurgeWorker, type: :worker do
+  describe ".perform_async" do
+    subject(:perform_async) { described_class.perform_async }
+
+    it "enqueues 1 job with expected options" do
+      expect { perform_async }
+        .to change(described_class, :jobs)
+          .from([])
+          .to(
+              [
+                hash_including(
+                  "retry" => true,
+                  "queue" => "default",
+                  "args" => [],
+                  "class" => "PurgeWorker"
+                )
+              ]
+          )
+    end
+  end
+
+  describe "#perform" do
+    subject(:perform) { described_class.new.perform }
+
+    let(:bulk_submission) { create(:bulk_submission, :with_original_file, :with_result_file, expires_at:) }
+    let(:submission) do
+ create(:submission, first_name: 'Rosie', last_name: 'Conway', nino: 'JC654321A', dob: '1977-03-08'.to_date,  
+bulk_submission:) end
+    let(:expires_at) { Time.zone.now }
+
+    before do
+      bulk_submission
+      submission
+    end
+
+    it_behaves_like "applcation worker logger"
+
+    context "when the bulk submission has expired" do
+      it "purges the original file" do
+        perform
+        bulk_submission.reload
+        expect(bulk_submission.original_file.attachment).to be_nil
+        expect(bulk_submission.original_file.blob).to be_nil
+      end
+  
+      it "purges the result file" do
+        perform
+        bulk_submission.reload
+        expect(bulk_submission.result_file.attachment).to be_nil
+        expect(bulk_submission.result_file.blob).to be_nil
+      end
+
+      it "updates the submission record" do
+        perform
+        submission.reload
+        expect(submission.first_name).to eq 'purged'
+        expect(submission.last_name).to eq 'purged'
+        expect(submission.nino).to eq 'AB123456C'
+        expect(submission.dob).to eq '01-01-1970'
+      end
+    end
+
+    context "when the bulk submission has not expired" do
+      let(:expires_at) { Time.zone.tomorrow }
```

Comment:
We should probably test the boundaries with a life like test that freezes time at 8pm - or whenever it is scheduled - and then expects the one that has expires_at for previous day/midnight to be purged/cleaned/removed but one with expires_at today at midnight to no be purged.
```suggestion
      let(:expires_at_yesterday) { 1.day.ago.midnight }
      let(:expires_at_today) { Time.current.midnight }
```
I wonder whether we need to also change functionality and test to cover the very edges
```
      let(:expires_at_yesterday) { 1.day.ago.midnight }
      let(:expires_at_this_am) { 1.day.ago.midnight + 1.second }
      let(:expires_at_future) { Time.current.midnight }
```

if we expire something manually at 13:05 what do we expect it to do? (open question)

Reply:
see https://github.com/ministryofjustice/laa-assure-hmrc-data/pull/280/commits/394c7764ff6f4168b1b1f098362d5e6c303f5e7c